src/main/java/nextstep/courses/domain/session/Enrollment.java

```diff
@@ -32,6 +43,44 @@ public Long getSessionId() {
         return sessionId;
     }
 
+    public void updateEnrollmentStatus(EnrollmentStatus enrollmentStatus) {
+        validateEnrollmentStatus(enrollmentStatus);
+        this.enrollmentStatus = enrollmentStatus;
+    }
+
+    private void validateEnrollmentStatus(EnrollmentStatus enrollmentStatus) {
+        validateNotSeletedAndApproved(enrollmentStatus);
+        validateSelectedAndCancelled(enrollmentStatus);
+    }
+
+    private void validateSelectedAndCancelled(EnrollmentStatus enrollmentStatus) {
+        if (isSelectedAndCancelled(enrollmentStatus)) {
+            throw new IllegalArgumentException("선발된 인원은 강의 취소가 불가능합니다.");
+        }
+    }
+
+    private void validateNotSeletedAndApproved(EnrollmentStatus enrollmentStatus) {
+        if (isNotSelectedAndApproved(enrollmentStatus)) {
+            throw new IllegalArgumentException("미선발된 인원은 강의 승인이 불가능합니다.");
+        }
+    }
+
+    private boolean isSelectedAndCancelled(EnrollmentStatus enrollmentStatus) {
+        return this.selectionStatus.equals(SelectionStatus.SELECTED) && enrollmentStatus.equals(EnrollmentStatus.CANCELLED);
```

Comment:
```suggestion
        return this.selectionStatus.isSelected() && enrollmentStatus.isCancled();
```
enum도 class와 같이 메시지를 보내는 방식으로 구현 가능함